pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```
@@ -1001,9 +1008,9 @@ public static String writeRequest(@NonNull String instanceId, @NonNull String dd
                 } catch (Exception e) {
                     //This error will trigger on studies with no participants, this skips
                     //the error log if that is the reason for the upsert failure.
-                        logger.error(String.format("Error inserting newly created kit request shipping with dsm kit request id: %s in "
+                    logger.error(String.format("Error inserting newly created kit request shipping with dsm kit request id: %s in "
```

It feels like this method should be more generic, and this error handling should be in the caller. (The method comment says it is called by a chron job, but this can easily be decoupled from that. IOW, this will be easier to test/verify if it does not make assumptions about the callers.)

---

To my understanding, this method takes care of writing the new kit ( from any study) into DB, which needs a number of queries to execute in one transaction, hence the length of the method, and then, if db update and insertion was successful, we also update ES. 

Looking at the history of this file, this comment was made for Darwin's arc because it does not have an es participants index. However, now that I have added a check for juniper ( and juniper-like studies that do not have an es participants index, which is also true for Darwin's) that comment is not valid anymore and this catch is only entered through the point of a real ES update failure. 

I may be missing the point of handling the error in the caller here not that the comment is not valid anymore. Please let me know if I am not looking at it the right way

---

Sorry, my comment was not clear.
I'm concerned about converting the exception to a log statement. Generally we would only do that at the interface point -- for example, http request handling, or pusbsub message handling, etc. This method appears to be called from a handful of different places, including a route. In that case the right thing to is let the callers handle the exception as they see fit, rather than swallowing it here.
I realize the form of this code already existed, and this is an opportunity for improvement. But I appreciate that is not technically in scope for this PR.

---

I see what you mean... I completely agree... My personal thought is that would go beyond the scope of Juniper status route... However, if you think this is needed I can make the changes. I just don't think changing all of those kit-related classes will benefit this specific PR. How about I make a tech-debt ticket and address them in there?

---

Sounds good.